jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/sender/PushNotificationSenderEndpoint.java

```diff
@@ -128,4 +180,21 @@ private PushApplication loadPushApplicationWhenAuthorized(HttpServletRequest req
         // unauthorized...
         return null;
     }
+
+    private void processPushMessages(final HttpServletRequest request, final PushApplication pushApplication, final List<InternalUnifiedPushMessage> messages) {
+
+        // read some metadata about the submitting client
+        final String ipAddress = HttpRequestUtil.extractIPAddress(request);
+        final String clientSender = HttpRequestUtil.extractAeroGearSenderInformation(request);
+
+        // apply client metada
+        for (InternalUnifiedPushMessage msg : messages) {
+            msg.setIpAddress(ipAddress);
+            msg.setClientIdentifier(clientSender);
```

Comment:
Here, why don't use Java8 to do this interaction and send here the message directly for don't be required do the same for again?